cmake: extensions: ToolchainCapabilityDatabase.cmake parse error in shippable

Testing PR #6760 on shippable produces an error that seems not related to the PR.

    CMake Error at /home/buildslave/.cache/zephyr/ToolchainCapabilityDatabase.cmake:405:
     Parse error.  Function missing ending ")".  End of file reached.
    Call Stack (most recent call first):
      ../../../cmake/extensions.cmake:620 (include)
      ../../../cmake/extensions.cmake:1002 (zephyr_check_compiler_flag)
      ../../../cmake/extensions.cmake:978 (target_cc_option_fallback)
      ../../../cmake/extensions.cmake:106 (target_cc_option)
      ../../../arch/arc/soc/em7d/CMakeLists.txt:6 (zephyr_cc_option)

For the full log see https://app.shippable.com/github/zephyrproject-rtos/zephyr/runs/14800/2/console

@SebastianBoe 
I did a fast update of the PR, presumably during it was tested by shippable. Maybe this caused a corruption of the cache file. Currently I can not retest as "recheck" seems not to work for me. 

Here a copy from the shippable console (just in case):
"""
  build_ci 1m 19s
  export CCACHE_DIR=${SHIPPABLE_BUILD_DIR}/ccache/.ccache 0s  
  if [ "$IS_PULL_REQUEST" = "true" ]; then 1s  
  source zephyr-env.sh 0s  
  ccache -c -s --max-size=2000M 1s  
  if [ "$MATRIX_BUILD" = "5" -a "$IS_PULL_REQUEST" = "true" ]; then 0s  
  if [ "$IS_PULL_REQUEST" = "true" ]; then 0s  
  ./scripts/sanitycheck ${SANITYCHECK_OPTIONS} --save-tests test_file.txt 1m 17s  
Selecting default platforms per test case
Building testcase defconfigs...
Could not build defconfig for em_starterkit_em7d_tests_kernel_fatal_kernel.common.stack_protection_config-sanitycheck
CMake Deprecation Warning at /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/cmake/app/boilerplate.cmake:38 (cmake_policy):
  The OLD behavior for policy CMP0000 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  CMakeLists.txt:1 (include)


-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.5.2", minimum required is "3.4") 
-- Selected BOARD em_starterkit_em7d
Zephyr version: 1.11.99
Parsing Kconfig tree in /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/Kconfig
Using /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/boards/arc/em_starterkit/em_starterkit_em7d_defconfig as base
Merging prj.conf
-- Generating zephyr/include/generated/generated_dts_board.h
-- Cache files will be written to: /home/buildslave/.cache/zephyr
-- The C compiler identification is GNU 6.2.1
-- The CXX compiler identification is GNU 6.2.1
-- The ASM compiler identification is GNU
-- Found assembler: /opt/sdk/zephyr-sdk-0.9.2/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/arc-zephyr-elf-gcc
-- Performing Test toolchain_is_ok
-- Performing Test toolchain_is_ok - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/em_starterkit_em7d/tests/kernel/fatal/kernel.common.stack_protection
make[1]: Entering directory '/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/em_starterkit_em7d/tests/kernel/fatal/kernel.common.stack_protection'
/usr/local/bin/cmake -H/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/tests/kernel/fatal -B/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/em_starterkit_em7d/tests/kernel/fatal/kernel.common.stack_protection --check-build-system CMakeFiles/Makefile.cmake 0
Re-run cmake file: Makefile older than: /home/buildslave/.cache/zephyr/ToolchainCapabilityDatabase.cmake
CMake Deprecation Warning at /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/cmake/app/boilerplate.cmake:38 (cmake_policy):
  The OLD behavior for policy CMP0000 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
Call Stack (most recent call first):
  CMakeLists.txt:1 (include)


-- Selected BOARD em_starterkit_em7d
Zephyr version: 1.11.99
Parsing Kconfig tree in /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/Kconfig
Using /home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/em_starterkit_em7d/tests/kernel/fatal/kernel.common.stack_protection/zephyr/.config as base
-- Generating zephyr/include/generated/generated_dts_board.h
-- Cache files will be written to: /home/buildslave/.cache/zephyr
CMake Error at /home/buildslave/.cache/zephyr/ToolchainCapabilityDatabase.cmake:405:
  Parse error.  Function missing ending ")".  End of file reached.
Call Stack (most recent call first):
  ../../../cmake/extensions.cmake:620 (include)
  ../../../cmake/extensions.cmake:1002 (zephyr_check_compiler_flag)
  ../../../cmake/extensions.cmake:978 (target_cc_option_fallback)
  ../../../cmake/extensions.cmake:106 (target_cc_option)
  ../../../arch/arc/soc/em7d/CMakeLists.txt:6 (zephyr_cc_option)


-- Configuring incomplete, errors occurred!
See also "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/em_starterkit_em7d/tests/kernel/fatal/kernel.common.stack_protection/CMakeFiles/CMakeOutput.log".
See also "/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/em_starterkit_em7d/tests/kernel/fatal/kernel.common.stack_protection/CMakeFiles/CMakeError.log".
Makefile:812: recipe for target 'cmake_check_build_system' failed
make[1]: *** [cmake_check_build_system] Error 1
make[1]: Target 'config-sanitycheck' not remade because of errors.
make[1]: Leaving directory '/home/buildslave/src/github.com/zephyrproject-rtos/zephyr/sanity-out/em_starterkit_em7d/tests/kernel/fatal/kernel.common.stack_protection'
Traceback (most recent call last):
  File "./scripts/sanitycheck", line 2875, in <module>
    main()
  File "./scripts/sanitycheck", line 2740, in main
    discards = ts.apply_filters()
  File "./scripts/sanitycheck", line 1890, in apply_filters
    raise SanityRuntimeError("Couldn't build some defconfigs")
__main__.SanityRuntimeError: Couldn't build some defconfigs
"""

--- a/.shippable.yml
+++ b/.shippable.yml
@@ -72,7 +72,7 @@ build:
       - rm test_file.txt
       - ccache -s
     on_failure:
-      - rm -rf ccache
+      - rm -rf ccache $HOME/.cache/zephyr
       - mkdir -p shippable/testresults
       - mkdir -p shippable/codecoverage
       - source zephyr-env.sh
@@ -97,7 +97,7 @@ build:
             cp ./scripts/sanity_chk/last_sanity.xml shippable/testresults/;
           fi;
     on_success:
-      - rm -rf ccache
+      - rm -rf ccache $HOME/.cache/zephyr
       - mkdir -p shippable/testresults
       - mkdir -p shippable/codecoverage
       - source zephyr-env.sh

